### PR TITLE
Solution_OOP_Single_Inheritance

### PR DESCRIPTION
**Pull request by IraMladanovych02**

### CODE REVIEW

File: tests/__init__.py
Comment (Dimosphen1) [inline tests/__init__.py:1]: There is no need to delete this file, remove deletion from the PR, please

File: tests/test_main.py
Comment (Dimosphen1) [inline tests/test_main.py:1]: There is no need to delete this file, remove deletion from the PR, please

File: app/main.py [hunk 1]
```diff
@@ -1 +1,41 @@
...
+
+    @staticmethod
+    def bite(animal):
+        if isinstance(animal, Herbivore) and not animal.hidden:
+            animal.decrease_health(50)
+
+
+pantera = Carnivore("Bagira")
+snake = Carnivore("Kaa")
+print(Animal.alive)
```
Comment (Dimosphen1) [inline app/main.py:41]: Remove these lines, there is no need to have them here

File: app/main.py [hunk 2]
```diff
@@ -1 +1,41 @@
...
+        super().__init__(name=name)
+
+    def hide(self):
+        self.hidden = not self.hidden
+class Carnivore(Animal):
+    def __init__(self, name):
+        super().__init__(name=name)
+
+    @staticmethod
+    def bite(animal):
```
Comment (Dimosphen1) [inline app/main.py:34]: Add type annotation to all methods

File: app/main.py [hunk 3]
```diff
@@ -1 +1,41 @@
...
+    def die(self):
+        Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def __init__(self, name):
+        super().__init__(name=name)
+
+    def hide(self):
+        self.hidden = not self.hidden
```
Comment (Dimosphen1) [inline app/main.py:28]: Add blank lines between different classes

File: app/main.py [hunk 4]
```diff
@@ -1 +1,41 @@
...
+            self.health = 0
+            self.die()
+
+    def die(self):
+        Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def __init__(self, name):
+        super().__init__(name=name)
```
Comment (Dimosphen1) [inline app/main.py:25]: There is no need to duplicate `__init__` in this class and in the `Carnivore` class as well, remove them and rely on the parent class implementation

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested. Make sure that all checks pass

**Review by pavlopro — CHANGES_REQUESTED** [review]
Tests Failed

**Pull request closed without merging**